src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityAssemblyLine.java

```diff
@@ -60,4 +74,128 @@ public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
     protected IBlockState getCasingState() {
         return MetaBlocks.METAL_CASING.getState(BlockMetalCasing.MetalCasingType.STEEL_SOLID);
     }
+
+    private int beamCount;
+    @SideOnly(Side.CLIENT)
+    private GTLaserBeamParticle[][] beamParticles;
+
+    @Override
+    public void update() {
+        super.update();
+        if(ConfigHolder.client.shader.assemblyLineParticles) {
+            if (getRecipeMapWorkable().isWorking()) {
+                int maxBeams = getAbilities(MultiblockAbility.IMPORT_ITEMS).size() + 1;
```

Comment:
Would it be better to pre-calculate this in `onStructureFormed()`, and save it as an instance variable and to NBT/syncData?

Reply:
I don't really see a point to saving it to NBT, because then we would have to check if this value is present in every call, because this config option can be switched off and on without having to restart the game. In addition, we do not need this value unless the config is enabled, so it doesn't really make sense to always calculate and write to NBT.